trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/http/HttpCredentialsProvider.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.credentials.http;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.inject.Inject;
+import io.airlift.http.client.FullJsonResponseHandler.JsonResponse;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.json.JsonCodec;
+import io.airlift.json.JsonCodecFactory;
+import io.airlift.log.Logger;
+import io.trino.aws.proxy.spi.credentials.Credentials;
+import io.trino.aws.proxy.spi.credentials.CredentialsProvider;
+import io.trino.aws.proxy.spi.credentials.Identity;
+import jakarta.ws.rs.core.UriBuilder;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static io.airlift.http.client.FullJsonResponseHandler.createFullJsonResponseHandler;
+import static io.airlift.http.client.Request.Builder.prepareGet;
+import static java.util.Objects.requireNonNull;
+
+public class HttpCredentialsProvider
+        implements CredentialsProvider
+{
+    private static final Logger log = Logger.get(HttpCredentialsProvider.class);
+
+    private final HttpClient httpClient;
+    private final JsonCodec<Credentials> jsonCodec;
+    private final URI httpCredentialsProviderEndpoint;
+
+    @Inject
+    public HttpCredentialsProvider(@ForHttpCredentialsProvider HttpClient httpClient, HttpCredentialsProviderConfig config, ObjectMapper objectMapper, Class<? extends Identity> identityClass)
+    {
+        requireNonNull(config, "Config is null");
```

Comment:
We're accessing a field in `config` already so we don't need the requireNonNull.

I'd however suggest using `requireNonNull` to check that `config.getEndpoint()` does not return null

Reply:
Since config used in the constructor, it's good to have the requireNonNull for config as well. 
The Config class already has the `@NonNull` for the endpoint - https://github.com/pranavr12/aws-proxy/blob/http-creds/trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/http/HttpCredentialsProviderConfig.java#L25.

Reply:
We don't usually add requireNonNull on configs.

Reply:
Removed requireNonNull on config